### DISCUSSION TITLE
Get index for k-hop neighbors separately using neighbor loader

### DISCUSSION THREAD

**Discussion by RX28666** [pyg-team/pytorch_geometric#6702] [Q&A]
Hello,

I am using NeighborLoader to do sampling, and using `data.n_id = torch.arange(data.num_nodes)` to get index for targeted nodes and their neighbors.
I am wondering if there is a way to get the index for one hop, second hop, etc. respectively. So I am able to know which nodes are the first/second hop neighbor of the targeted nodes.
Thanks.

**Comment by LukeLIN-web** [comment]
I also have this problem.

**Reply by rusty1s** [reply]
We are working on integrating this here: https://github.com/pyg-team/pyg-lib/pull/197
